.github/workflows/test-env-deploy.yml

```
@@ -165,12 +164,39 @@ jobs:
               for(let i=1; i<=numberOfContainers[1]; i++){
                 containers.push(i)
               }
+            echo ${{Array.from(containers.keys())}}
             return {"containers": containers}
 
       - name: echo-tags
         run: |
           echo ${{steps.get_tags.outputs.result}}
 
+  testmo-report-preparation:
+    needs: prepare-tests
+    runs-on: ubuntu-latest
+    outputs:
+      testmo-run-id: ${{ steps.run-tests.outputs.testmo-run-id }}
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 19
+          cache: npm
+      - run: npm ci
+      - run: |
+          npx testmo automation:run:create \
+            --instance "$TESTMO_URL" \
+            --project-id 1 \
+            --name "Deployment tests" \
+            --resources resources.json \
+            --source "unit-tests" > testmo-run-id.txt
+          ID=$(cat testmo-run-id.txt)
+          echo "testmo-run-id=$ID" >> $GITHUB_OUTPUT
+        env:
+          TESTMO_URL: ${{ secrets.TESTMO_URL }}
+          TESTMO_TOKEN: ${{ secrets.TESTMO_TOKEN }}
+        id: run-tests
+
   cypress-run-selected:
     runs-on: ubuntu-latest
     needs: [prepare-tests, deploy]
```

Aren't you missing this?

```suggestion
    needs: [prepare-tests, deploy, testmo-report-preparation]
```

---

Fixed